src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirModel.cs

```diff
@@ -47,94 +49,98 @@ public sealed class SqlServerFhirModel : SqlServerModelInitializer
 
         public SqlServerFhirModel(
             SqlServerDataStoreConfiguration configuration,
-            SchemaInformation schemaInformation,
-            ISearchParameterDefinitionManager.SupportedSearchParameterDefinitionManagerResolver searchParameterDefinitionManagerResolver,
+            SchemaInitializer schemaInitializer,
+            ISearchParameterDefinitionManager searchParameterDefinitionManager,
             IOptions<SecurityConfiguration> securityConfiguration,
             ILogger<SqlServerFhirModel> logger)
-            : base(configuration, schemaInformation, logger)
         {
-            EnsureArg.IsNotNull(searchParameterDefinitionManagerResolver, nameof(searchParameterDefinitionManagerResolver));
+            EnsureArg.IsNotNull(configuration, nameof(configuration));
+            EnsureArg.IsNotNull(schemaInitializer, nameof(schemaInitializer));
+            EnsureArg.IsNotNull(searchParameterDefinitionManager, nameof(searchParameterDefinitionManager));
             EnsureArg.IsNotNull(securityConfiguration?.Value, nameof(securityConfiguration));
             EnsureArg.IsNotNull(logger, nameof(logger));
 
-            _searchParameterDefinitionManager = searchParameterDefinitionManagerResolver();
+            _configuration = configuration;
+            _schemaInitializer = schemaInitializer;
+            _searchParameterDefinitionManager = searchParameterDefinitionManager;
             _securityConfiguration = securityConfiguration.Value;
             _logger = logger;
         }
 
         public short GetResourceTypeId(string resourceTypeName)
         {
-            ThrowIfNotInitialized();
             return _resourceTypeToId[resourceTypeName];
         }
 
         public bool TryGetResourceTypeId(string resourceTypeName, out short id)
         {
-            ThrowIfNotInitialized();
             return _resourceTypeToId.TryGetValue(resourceTypeName, out id);
         }
 
         public string GetResourceTypeName(short resourceTypeId)
         {
```

Comment:
ThrowIfNotInitialized() will fail in a predicatable way with a better error message than something like a null ref

Reply:
I was under the impression that these methods would never be called before `Start()` and `StartAsync()`, so there wouldn't be a need to check initialization. Please correct me if I'm wrong!

Reply:
Start _should_ be called first. This is just contract enforcement. The class cannot be used until Started, which makes it special, and it will tell you if try to use it before Started (i.e. in a unit test). (Or if ever we did publish this a general-purpose library, though I'm contradicting my comment on ConfigureAwait :)) Not a big deal either way.

Reply:
You're right - I found this useful when debugging unit tests. Added the check back in in c583087.